docs/glossary.md

```diff
@@ -0,0 +1,62 @@
+---
+id: glossary
+title: "Glossary of Terms"
+---
+**Batch:** A selection of records from a Data Asset.
+
+**Batch Request:** Provided to a Datasource in order to create a Batch.
+
+**CLI:** Command Line Interface
+
+**Catalog Asset:** A collection of records within a Datasource which is usually named based on the underlying data system and sliced in different ways to correspond to multiple desired specifications.
+
+**Checkpoint:** The primary means for validating data in a production deployment of Great Expectations.
+
+**Custom Expectation:** An extension of the `Expectation` class, developed outside of the Great Expectations library.
+
+**Data Asset:** A collection of records within a Datasource which is usually named based on the underlying data system and sliced to correspond to a desired specification.
+
+**Data Connector:** Provides the configuration details based on the source data system which are needed by a Datasource to define Data Assets.
+
+**Data Context:** The primary entry point for a Great Expectations deployment, with configurations and methods for all supporting components.
+
+**Data Docs:** Human readable documentation generated from Great Expectations metadata detailing Expectations, Validation Results, etc.
+
+**Datasource:** Provides a standard API for accessing and interacting with data from a wide variety of source systems.
```

Comment:
the "standard API" thing doesn't really make sense to our user. I'd say "Used to access and interact with data"; it might be worth mentioning that it connects a Connector to an Execution Engine

Reply:
In this case, the fact that the Datasource API remains the same regardless of the source data system it interacts with is one of the core promises that we want to emphasize -- that's what the "standard API" is supposed to indicate: that you don't have to change how you interact with your Datasource if you, for example, switch between one working with a filesystem to one working with a database.